test/models/nic_test.rb

```diff
@@ -386,7 +386,7 @@ def teardown
       end
 
       test "provision flag is set for primary interface automatically" do
-        primary = FactoryBot.build(:nic_managed, :primary => true, :provision => false,
+        primary = FactoryBot.build(:nic_managed, :primary => true, :provision => true,
                                     :domain => FactoryBot.build(:domain))
```

Comment:
> @ares currently we have a method set_provisioning_flag which sets automatically provision flag during validation for a NIC that is non-primary.

I think you misread the code.

```ruby
# in interface model
    def set_provisioning_flag
      return unless primary?               # non primary interfaces are skipped
      return unless host.present?       # ignore cases where this interface does not have host set yet
      self.provision = true if host.interfaces.detect(&:provision).nil?  # set this interface (only primary thanks to first line) as provision if and only if none host interfaces is flagged as provision yet
    end
```
This method sets the provisioning flag for **primary** interface if there's not any other interface with provisioning flag for a host. Meaning if user does not say otherwise, primary becomes also provisioning. In UI this happens automatically, but in API/cloning, that's a good thing.

Reply:
Ok @dhlavac I suggest to keep this behavior then.